cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/validator/CoreValidator.java

```diff
@@ -0,0 +1,311 @@
+     */
+    public static boolean maxValue(long val, long maxVal) {
+      return GenericValidator.maxValue(val, maxVal);
+    }
+
+    /**
+     * Checks if the value is less than or equal to the max.
+     */
+    public static boolean maxValue(int val, int maxVal) {
+      return GenericValidator.maxValue(val, maxVal);
+    }
+
+    /**
+     * Checks if the value is less than or equal to the max.
+     */
+    public static boolean maxValue(float val, float maxVal) {
+      return GenericValidator.maxValue(val, maxVal);
+    }
+
+    /**
+     * Checks if the value is greater than or equal to the min.
+     */
+    public static boolean minValue(double val, double minVal) {
+      return GenericValidator.minValue(val, minVal);
+    }
+
+    /**
+     * Checks if the value is greater than or equal to the min.
+     */
+    public static boolean minValue(long val, long minVal) {
+      return GenericValidator.minValue(val, minVal);
+    }
+
+    /**
+     * Checks if the value is greater than or equal to the min.
+     */
+    public static boolean minValue(int val, int minVal) {
+      return GenericValidator.minValue(val, minVal);
+    }
+
+    /**
+     * Checks if the value is greater than or equal to the min.
+     */
+    public static boolean minValue(float val, float minVal) {
+      return GenericValidator.minValue(val, minVal);
+    }
+
+    /**
+     * Checks if the value's adjusted length is greater than or equal to the min.
+     */
+    public static boolean minLength(String input, int length) {
+      return GenericValidator.minLength(input, length);
+    }
+
+    /**
+     * Checks if the code is either a valid ISBN-10 or ISBN-13 code.
+     */
+    public static boolean isValidISBN(String isbn) {
+      return ISBN_VALIDATOR.isValid(isbn);
+    }
+
+    /**
+     * Validates an IPv4 address.
+     */
+    public static boolean isValidInet4Address(String ipv4) {
+      return INET_ADDRESS_VALIDATOR.isValidInet4Address(ipv4);
+    }
+
+    /**
+     * Validates an IPv6 address.
+     */
+    public static boolean isValidInet6Address(String ipv6) {
+      return INET_ADDRESS_VALIDATOR.isValidInet6Address(ipv6);
+    }
+
+    /**
+     * Checks if the specified string is a valid IP address.
+     */
+    public static boolean isValidIp(String ip) {
+      return INET_ADDRESS_VALIDATOR.isValid(ip);
+    }
+
+    /**
+     * Returns true if the specified String matches any IANA-defined country code top-level domain.
+     */
+    public static boolean isValidCountryCodeTid(String ccTld) {
+      return DOMAIN_VALIDATOR.isValidCountryCodeTld(ccTld);
+    }
+
+    /**
+     * Returns true if the specified String matches any IANA-defined generic top-level domain.
+     */
+    public static boolean isValidGenericTId(String gTld) {
+      return DOMAIN_VALIDATOR.isValidGenericTld(gTld);
+    }
+
+
+    /**
+     * Returns true if the specified String matches any IANA-defined infrastructure top-level domain.
```

Comment:
Checks if the specified String matches an IANA-defined infrastructure top-level domain.